orangecontrib/text/widgets/owconcordance.py

```diff
@@ -0,0 +1,218 @@
+from collections import Counter
+from typing import Optional
+
+from AnyQt.QtCore import Qt, QAbstractTableModel, QSize
+from AnyQt.QtWidgets import QSizePolicy, QApplication, QTableView
+from AnyQt.QtGui import QBrush, QColor
+
+
+from Orange.data import Table
+from Orange.widgets import gui
+from Orange.widgets.settings import Setting
+from Orange.widgets.widget import OWWidget
+from orangecontrib.text.corpus import Corpus
+from orangecontrib.text.preprocess import Preprocessor, WordPunctTokenizer
+
+
+class IO:
+    DATA = 'Corpus'
+
+
+class ConcordanceModel(QAbstractTableModel):
+    "A model for constructing concordances from text."
+    def __init__(self):
+        QAbstractTableModel.__init__(self)
+        self.word = None
+        self.data = None
+        self.indices = None
+        self.doc_count = None
+
+    def set_word(self, word):
+        self.modelAboutToBeReset.emit()
+        self.word = word
+        self._compute_indices()
+        self.modelReset.emit()
+
+    def set_data(self, data):
+        self.modelAboutToBeReset.emit()
+        self.data = data
+        self._compute_indices()
+        self.modelReset.emit()
+
+    def flags(self, _):
+        return Qt.ItemIsEnabled | Qt.ItemIsSelectable
+
+    def rowCount(self, parent=None, *args, **kwargs):
+        return len(self.doc_count) if self.doc_count is not None else 0
+        # return len(self.indices) if self.indices is not None else 0
```

Comment:
No, these are the latest transformations I made. I intended for one row to be concordances from one document. Thus when the user selects the row, the corresponding document would be output. Would this not make the widget more useful?